### PR TITLE
add interactive examples for javascript

### PR DESCRIPTION
**Pull request by dhruvdutt**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,6 @@ bower_components
 
 # Ignore public dir - auto-generated on build
 public/*
+
+# search index
+src/data/search.index.json
```
Comment (dhruvdutt) [inline .gitignore]: @divyanshu013 This will be auto-generated, right?

File: content/docs/interactive/javascript.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+<iframe height="600px" width="100%" src="https://repl.it/@dhruvdutt/Appbaseio-Nodejs-searchStream-index?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+<!-- Mirage: stream query note? -->
+
+
+## GeoDistance Query
+
+TODO: Desc
+
+Get all the properties around a particular (lat,lon) co-ordinate.
```
Comment (dhruvdutt) [inline content/docs/interactive/javascript.md]: Tried [this](https://opensource.appbase.io/mirage/#?input_state=XQAAAAKmCAAAAAAAAAA9iIhnNAWbsswtYjeQNZkpzQK4_mOzUeDpWmI_q1W5J_v7Zsy4Ujaaw71A1BS9rYYbaidH1ngBtQ-I1sDSXRgrmGsCDzBYBoUXwDHQtefpH-PChYyyKqpnVdVrmIsxvIDhOBtThtu_W53GnLmSMoma1UPnh9E7LZRkgXxp3ltXA31wX1fcfowk1r2gVrCN8VgmuPFOWM3o65_HcKkYs4OQ0hAB7hnHy3CILQ5MgAbYZpuCAVHzQcRXBvN2fFZCuCSUNYX32cshZ3d4lrzfi6fQXgJyuLD2vgQns3dUESCtKWJx7m6L3fxjViX04mcxbte-NrsTOVNx3X2YvOd2UG_O3P5pgvLecXNPgnsZACPDmsU8OKslnavslVZEjTn_TFj6xc7bgYqdvYkOK1-UKMgOKP0J3jYoVLUimxuXXxXt6dP_ra5LFnfLugV6q2FimZ68iEFmUQu3DhAd_k8ND5N0bJvuBck1dR1iEAz6RDOVmkEhoPiChWYu13ode1Gg0LaGwxR7f5Sw6bXH7iQVo-tF9ZNLgxWzDusXVR1LhIE5FDRzAi-sirhi_O1gVKDhgz56fNmhVGQA26KqVbqsJxF9kCYb8gLRUCMdX5Taj4cuL7Z9JlWm7slNgYLW82a5MXksCXeFSNJ_p5wqkkf5pUCEx4S5xJ-ZW_ukRSv6opCd5QrbnYAer2xySyVSwlqEWicJYXluRI0aHNWfOyZxdEI_XqCkXLAJ9_wIQj1PC-vLwaaHJ2CAQ3algt-iMjUc4E6oR3IxBxH-VZ8p9_OCTVR311kTAwoKVKUyNpGDoc_uhjOTIdM5DDB8uQI5VkQ5Zrt-y81p8c5qM_90vYYA) on Mirage but not entirely sure what we need here.

Can you please link me an example query?

File: content/docs/interactive/javascript.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+<br/>
+
+TODO: REPL
+
+## Full-text Search
+
+Let's now do a [full-text search query](https://www.elastic.co/guide/en/elasticsearch/reference/current/full-text-queries.html) on the `name` field.
+Here, we'll use appbase-js [`search()`](https://docs.appbase.io/javascript/api-reference.html#search) method to fire a [match query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html) with our field `name` and query value.
+
```
Comment (dhruvdutt) [inline content/docs/interactive/javascript.md]: TODO: ngram type and explaining how ngrams work.

File: content/docs/interactive/javascript.md (hunk 2 above)
Comment (siddharthlatest) [reply content/docs/interactive/javascript.md]: Something like [this](https://opensource.appbase.io/mirage/#?input_state=XQAAAAK0CAAAAAAAAAA9iIhnNAWbsswtYjeQNZkpzQK4_mOzUeDpWmI_q1W5J_v7Zsy4Ujaaw71A1BS9rYYbaidH1ngBtQ-I1sDSXRgrmGsCDzBYBoUXwDHQtefpH-PChYyyKqpnVdVrmIsxvIDhOBtThtu_W53GnLmSMoma1UPnh9E7LZRkgXxp3ltXA31wX1fcfowk1r2gVrCN8VgmuPFOWM3o65_HcKkYs4OQ0hAB7hnHy3CILQ5MgAbYZpuCAVHzQcRXBvN2fFZCuCSUNYX32cshZ3d4lrzfi6fQXgJyuLD2vgQns3dUESCtKWJx7m6L3fxjViX04mcxbte-NrsTOVNx3X2YvOd2UG_O3P5pgvLecXNPgnsZACPDmsU8OKslnavslVZEjTn_TFj6xc7bgYqdvYkOK1-UKMgOKP0J3jYoVLUimxuXXxXxgHLnuSBlYv4MbQi1Hw9WjzpILKsFOhJNVGk3h88itDk2RILhqrJIU74eps4pG2P_9U-HhoWZ4vJXrsMx4sgFLKwuxL9JnpcoI8I9h3KCvxj2JVBPKrS2EACCLhtc-_1rz1FVJZZFtuHP4mvbmoYOWEpxOb6adSXi4Pf2DMuwPSBUTDVUc0JFmAgbj3uc87aSFzvbyhabjxPcjthDQCIsXfXwVAEr_-T-YQ1Q_VMvfGlPPq-k6ze89d9Y593K_HoB7G2WX35A-ha3oTTld7jcHFuKjX_042jNudf8e2uIjQgbhAD9icDO55t5cbJk3t0zBzAAI7W9Yk8ggDY889WnR4tLL1aO669LO3QpZGOZzzLVTDTSurgTi9FKtwI5397AOTlz00ZTc_FeeH4l5syXeqGQnXoBRvcoFWau0mfdJfQin__2WFBm) and a distance added as `5mi`.

**Pull request merged**